catt/controllers.py

```diff
@@ -92,50 +63,81 @@ def setup_cast(device_name, video_url=None, prep=None, controller=None, ytdl_opt
     except (pychromecast.error.ChromecastConnectionError, ValueError):
         cast = get_chromecast(device_name)
         cache.set_data(cast.name, cast.host, cast.port)
+
     cast.wait()
+    return cast
 
-    if video_url:
-        model_name = DEVICES_WITH_TWO_MODEL_NAMES.get(cast.model_name, cast.model_name)
-        cc_info = (cast.device.manufacturer, model_name)
-        stream = StreamInfo(
-            video_url, model=cc_info, host=cast.host, device_type=cast.cast_type, ytdl_options=ytdl_options
-        )
 
-    if controller:
-        if controller == "default":
-            app = DEFAULT_APP
-        else:
-            app = next(a for a in APP_INFO if a["app_name"] == controller)
-    elif stream and prep == "app":
-        if stream.is_local_file:
-            app = DEFAULT_APP
+def get_stream(url, device_info=None, host=None, ytdl_options=None):
+    if device_info:
```

Comment:
Minor suggestion, when initializing variables and then overwriting them on some condition (vs initializing them two different ways), I think this is slightly clearer:

```suggestion
    cc_info = cast_type = None
    if device_info:
        model_name = DEVICES_WITH_TWO_MODEL_NAMES.get(device_info.model_name, device_info.model_name)
        cc_info = (device_info.manufacturer, model_name)
        cast_type = device_info.cast_type
```

Reply:
Cheers